java/c/src/main/java/org/apache/arrow/c/ArrayExporter.java

```diff
@@ -91,7 +91,8 @@ void export(ArrowArray array, FieldVector vector, DictionaryProvider dictionaryP
 
       if (buffers != null) {
         data.buffers = new ArrayList<>(buffers.size());
```

Comment:
Shouldn't this also use the new buffer count? And we shouldn't call getFieldBuffers above at all anymore

Reply:
Yes, that's right.